File: src/main/java/com/shinonometn/re/ssim/caterpillar/SpiderStatus.java
package com.shinonometn.re.ssim.caterpillar;

import java.util.Date;
import java.util.List;

public interface SpiderStatus {
    String getName();

    String getStatus();

    int getThread();

    int getTotalPageCount();

    int getLeftPageCount();

    int getSuccessPageCount();

    int getErrorPageCount();

    List<String> getErrorPages();

    void start();

    void stop();

    Date getStartTime();

    int getPagePerSecond();
}

File: re-ssim-caterpillar/src/main/java/com/shinonometn/re/ssim/service/caterpillar/SpiderStatus.kt
package com.shinonometn.re.ssim.service.caterpillar

import java.util.Date

interface SpiderStatus {
    val name: String

    val status: String

    val thread: Int

    val totalPageCount: Int

    val leftPageCount: Int

    val successPageCount: Int

    val errorPageCount: Int

    val errorPages: List<String>

    val startTime: Date

    val pagePerSecond: Int

    fun start()

    fun stop()
}
